backend/build.gradle

```diff
@@ -21,6 +21,8 @@ dependencies {
 	implementation 'org.springframework.boot:spring-boot-starter-web'
 	implementation 'org.springframework.boot:spring-boot-starter-data-mongodb'
 	implementation 'org.springframework.boot:spring-boot-starter-data-rest'
+	implementation 'org.mapstruct:mapstruct:1.5.3.Final'
+	annotationProcessor 'org.mapstruct:mapstruct-processor:1.5.3.Final'
```

Comment:
Ich kenne Mapstruct nicht, aber es sah auf den ersten Blick so aus als ob das sehr ähnlich zu Automapper bei C# ist. In dem Fall hätte ich dir dringend davon abgeraten, weil Automapper erst zur Runtime Fehler schmeist und nicht zur Compile Time. Hab aber mal ein bisschen dazu gegooglet und es sieht so aus als ob das bei Mapstruct anders gelöst wäre und du schon zur Compile Time Errors bekommst.

Mein Gefühl sagt mir trotzdem, dass ich eher die Mapper selbst schreiben würde. Das ist in der Regel schnell gemacht und hat keine große Komplexität, zeigt jedoch ein bisschen klarer was passiert. Ich muss aber zugeben, dass Mapstruct auch seinen Charme zu haben scheint. Mach es also gerne so wie es dir lieber ist. Wollte nur mal meine Gedanken teilen. ;)